default_error_overlay.yaml

```diff
@@ -0,0 +1,38 @@
+overlay: 1.0.0
```

Comment:
overlay to change the naming of the component error to default_error to avoid go language collisions

Reply:
Thanks for adding this overlay and showcasing this feature!
I have one question and one requested change before we merge this in.
1. Question: Is there a way to rename the field without re-defining its contents? I can see this as a flaky part of our API infrastructure if we have the error object defined in this language-specific overlay, as with my understanding, changes in the base field won't get automatically propagated here.
2. Request: change the new field name from default_error to generic-error.

Reply:
I actually thought of a much cleaner way to overlay this using our [name override extensions](https://www.speakeasyapi.dev/docs/openapi/best-practices#x-speakeasy-name-override)

This will keep all of your content. It just renames the error field in the SDK

Reply:
Thank you!!! Love the extension. I'm happy with this upgrade.